src/ion.hpp

```diff
@@ -37,116 +31,13 @@ inline std::string to_string(ionKind k) {
-        eX_{idx.size(), std::numeric_limits<value_type>::quiet_NaN()},
-        Xi_{idx.size(), std::numeric_limits<value_type>::quiet_NaN()},
-        Xo_{idx.size(), std::numeric_limits<value_type>::quiet_NaN()},
-        charge(0),
-        default_int_concentration(0),
-        default_ext_concentration(0)
-    {}
-
-    // Set the weights used when setting default concentration values in each CV.
-    // The concentration of an ion species in a CV is a linear combination of
-    // default concentration and contributions from mechanisms that update the
-    // concentration. The weight is a value between 0 and 1 that represents the
-    // proportion of the CV area for which the default value is to be used
-    // (i.e. the proportion of the CV where the concentration is prescribed by a
-    // mechanism).
-    void set_weights(const std::vector<value_type>& win, const std::vector<value_type>& wout) {
-        EXPECTS(win.size()  == size());
-        EXPECTS(wout.size() == size());
-        weight_Xi_ = memory::make_const_view(win);
-        weight_Xo_ = memory::make_const_view(wout);
-    }
-
-    view current_density() {
-        return iX_;
-    }
-
-    view reversal_potential() {
-        return eX_;
-    }
-
-    view internal_concentration() {
-        return Xi_;
-    }
-
-    view external_concentration() {
-        return Xo_;
-    }
-
-    view internal_concentration_weights() {
-        return weight_Xi_;
-    }
-
-    view external_concentration_weights() {
-        return weight_Xo_;
-    }
-
-    void reset(double temperature_K) {
-        // The Nernst equation uses the assumption of nonzero concentrations:
-        EXPECTS(default_int_concentration > value_type(0));
-        EXPECTS(default_ext_concentration > value_type(0));
-        memory::fill(iX_, 0); // reset current
-        init_concentration(); // reset internal and external concentrations
-        nernst_reversal_potential(temperature_K);
-    }
-
-    // Calculate the reversal potential for all compartments using Nernst equation.
-    // Temperature is in degrees Kelvin
-    void nernst_reversal_potential(value_type temperature_K) {
-        backend::nernst(charge, temperature_K, Xo_, Xi_, eX_);
-    }
-
-    void init_concentration() {
-        backend::init_concentration(
-            Xi_, Xo_, weight_Xi_, weight_Xo_,
-            default_int_concentration, default_ext_concentration);
-    }
-
-    const_iview node_index() const {
-        return node_index_;
-    }
-
-    std::size_t size() const {
-        return node_index_.size();
-    }
-
-private:
-    iarray node_index_;
-    array iX_;          // (nA) current
-    array eX_;          // (mV) reversal potential
-    array Xi_;          // (mM) internal concentration
-    array Xo_;          // (mM) external concentration
-    array weight_Xi_;   // (1) concentration weight internal
-    array weight_Xo_;   // (1) concentration weight external
+// Ion (species) description
```

Comment:
Move the comment that explains the legacy NEURON naming scheme for ion species variables to the fvm shared state whither it was refactored.